cmd/juju/commands/bootstrap.go

```diff
@@ -474,134 +372,19 @@ func (c *bootstrapCommand) Run(ctx *cmd.Context) (resultErr error) {
 		c.controllerName = defaultControllerName(c.Cloud, region.Name)
 	}
 
-	controllerModelUUID, err := utils.NewUUID()
-	if err != nil {
-		return errors.Trace(err)
-	}
-	hostedModelUUID, err := utils.NewUUID()
-	if err != nil {
-		return errors.Trace(err)
-	}
-	controllerUUID, err := utils.NewUUID()
-	if err != nil {
-		return errors.Trace(err)
-	}
-
-	// Create a model config, and split out any controller
-	// and bootstrap config attributes.
-	modelConfigAttrs := map[string]interface{}{
-		"type":         cloud.Type,
-		"name":         bootstrap.ControllerModelName,
-		config.UUIDKey: controllerModelUUID.String(),
-	}
-
-	userConfigAttrs, err := c.config.ReadAttrs(ctx)
-	if err != nil {
-		return errors.Trace(err)
-	}
-	modelDefaultConfigAttrs, err := c.modelDefaults.ReadAttrs(ctx)
+	bootstrapModelConfig,
+		controllerConfig,
+		bootstrapConfig,
+		inheritedControllerAttrs,
+		userConfigAttrs,
+		err := c.getBootstrapConfigs(ctx, cloud, provider)
```

Comment:
This seems a bit awkward It would be nicer to read if c.getBootstrapConfigs would return a struct containing all of these things and even more go idiomatic if it was called c.bootstrapConfigs